### ISSUE TITLE
similar to subcommand, I would like to have subarg

### ISSUE THREAD

**Issue by kernel164** [alexflint/go-arg#126]
Here is the idea...

```
type MainOpts struct {
	Port    int `arg:"--port"`
	Inner InnerOpts `arg:"subarg:inner"`
}
type InnerOpts struct {
	Enabled bool `arg:"enabled"`
}
```

should become

```
--port
--inner-enabled
```

Don't know whether it is already supported but this nicely fits in for nested structs.
It's similar to custom parsing for values but for arg names.

**Comment by alexflint** [comment]
Yes this is do-able but I fear that it would add more complexity than it's worth, and would make it more difficult to understand code that uses this library. Are there really cases when you want to repeat a significant number of inner options at different places within the same program?

**Comment by kernel164** [comment]
Our requirement is similar to this => https://doc.traefik.io/traefik/v2.3/reference/static-configuration/cli/

possibly there could be 3 or 4 levels of inner struct.

**Comment by kernel164** [comment]
we used kingpin => https://github.com/alecthomas/kingpin but we would like to move to go-arg.

Idea is to have go-arg,config(yaml,json.. etc) parse into same struct.

**Comment by alexflint** [comment]
Got it. Seems reasonable, and fun to implement :). I'm about to go into a meditation retreat but will try to find time for this afterwards.

**Comment by alexflint** [comment]
Would you be interested in providing some funding to me in order to implement this feature?

**Comment by kernel164** [comment]
I just do opensource projects like you. I started using go-arg for my new codes. thought of having this feature in my new projects. I can contribute as code though.

**Comment by alexflint** [comment]
OK @kernel164, I'd be happy to review a pull request
